feat: font-lock for `record_field`

dalugm opened the pull request:
Only for src like this:

```dart
var record = ('first', a: 2, b: true, 'last');
```

After this patch, `a` and `b` will be highlighted as `type`.

---

Code like this will not be highlighted:

```dart
var (i as int, s as String) = record;
```

tree-sitter-dart has error when parsing `i` and `s` for now, need to wait for upstream updates

dalugm committed:
feat: font-lock for `record_field`

50ways2sayhard commented:
> tree-sitter-dart has error when parsing i and s for now, need to wait for upstream updates

Are there any related issue or pr we can keep track with?

dalugm committed:
fix: font-lock for `property`

There is no face named `font-lock-property-face`, only
`font-lock-property-name-face` and `font-lock-property-use-face`.

dalugm commented:
> Are there any related issue or pr we can keep track with?

Sry, I hadn't check it, I only add the font-lock according to
`treesit-explorer-mode`.

When I'm using `treesit-explorer-mode`, the explorer buffer shows ERROR
when parsing the new record pattern.

---

Maybe we can follow it here:

https://github.com/UserNobody14/tree-sitter-dart/issues/42

I believe the pattern's support will come soon

dalugm commented:
The latest tree-sitter-dart's commit had added the support for type cast.

https://github.com/UserNobody14/tree-sitter-dart/commit/692401954ed047136c1b9d9925214f4703350ee7

And dart-ts-mode works fine with it, I guess there is no more work need to do for now.
